src/wrappers/themis/rust/src/secure_cell.rs

```diff
@@ -25,36 +91,85 @@ use bindings::{
-///
-/// * `seal()`
-/// * `token_protect()`
-/// * `context_imprint()`
+/// This is modeless, basic cell. First you create a `SecureCell` object with a key and an optional
+/// context then you select the desired operation mode and your Secure Cell is ready to go.
 pub struct SecureCell {
     master_key: KeyBytes,
     user_context: KeyBytes,
 }
 
 impl SecureCell {
     /// Constructs a new cell secured by a master key.
+    ///
+    /// # Examples
+    ///
+    /// A master key may be provided as anything convertible to a byte slice: a byte slice or an
+    /// array, a `Vec<u8>`, or a `String`.
+    ///
+    /// Note that it is actually _a password_. It is not required to be an actual key generated
+    /// by one of the [`keygen`] functions (though you definitely can use those keys as well):
+    ///
+    /// ```
+    /// use themis::secure_cell::SecureCell;
+    ///
+    /// SecureCell::with_key(b"byte string");
+    /// SecureCell::with_key(&[1, 2, 3, 4, 5]);
+    /// SecureCell::with_key(vec![6, 7, 8, 9]);
+    /// SecureCell::with_key(format!("owned string"));
+    /// ```
+    ///
+    /// This method is equivalent to [`with_key_and_context`] called with an empty context (`&[]`):
+    ///
+    /// ```
+    /// # fn main() -> Result<(), themis::Error> {
+    /// use themis::secure_cell::SecureCell;
+    ///
+    /// let cell1 = SecureCell::with_key(b"key").seal();
+    /// let cell2 = SecureCell::with_key_and_context(b"key", &[]).seal();
+    ///
+    /// let encrypted = cell1.encrypt(b"some data")?;
+    /// let decrypted = cell2.decrypt(&encrypted)?;
+    /// assert_eq!(decrypted, b"some data");
+    /// # Ok(())
+    /// # }
+    /// ```
+    ///
+    /// [`keygen`]: ../keygen/index.html
+    /// [`with_key_and_context`]: #method.with_key_and_context
     pub fn with_key<K: AsRef<[u8]>>(master_key: K) -> Self {
         Self {
             master_key: KeyBytes::copy_slice(master_key.as_ref()),
             user_context: KeyBytes::empty(),
         }
     }
 
-    /// Constructs a new cell secured by a master key and arbitrary user data.
+    /// Constructs a new cell secured by a master key and arbitrary “context information”.
     ///
     /// User context will be included into encrypted data in a manner specific to the operation
-    /// mode. See module-level documentation for details. You will need to provide this context
-    /// again in order to extract the original data from the cell.
+    /// mode. See [module-level documentation][secure_cell] for details. You will need to provide
+    /// this context again in order to extract the original data from the cell.
+    ///
+    /// # Examples
+    ///
+    /// As with the key, the context information can be anything convertible into a byte slice;
+    ///
+    /// ```
+    /// use themis::secure_cell::SecureCell;
+    ///
+    /// SecureCell::with_key_and_context(&[], b"byte string");
```

Comment:
can you change examples without usage of empty keys? themis doesn't allow to use empty keys:
* https://github.com/cossacklabs/themis/blob/8b877052f7c9384f52d81f6358882fe3a788cbd8/src/soter/openssl/soter_sym.c#L132
* https://github.com/cossacklabs/themis/blob/efe288ebaaababa1c7adc633e3f8cf325e3db0be/src/soter/boringssl/soter_sym.c#L132

plus constructor with context as parameter force to use one context for all encryptions but more securely when user will use different contexts for separate encryptions. all other our wrappers don't provide to initialize secure cell component with pre-defined context. better to force using different contexts with extra parameter in `encrypt`/`seal` method

Reply:
Thank you, I see a commit with updated keys in examples.

Regarding context – ok, will wait for next PR.